crates/wasmtime/src/store.rs

```diff
@@ -614,6 +614,29 @@ impl<T> Store<T> {
         self.inner.add_fuel(fuel)
     }
 
+    /// Synthetically consumes fuel from this [`Store`].
+    ///
+    /// For this method to work fuel consumption must be enabled via
+    /// [`Config::consume_fuel`](crate::Config::consume_fuel).
+    ///
+    /// WebAssembly execution will automatically consume fuel but if so desired
+    /// the embedder can also consume fuel manually to account for relative
+    /// costs of host functions, for example.
+    ///
+    /// This function will attempt to consume `fuel` units of fuel from within
+    /// this store. If the remaining amount of fuel allows this then `Ok(N)` is
+    /// returned where `N` is the amount of remaining fuel. Otherwise an error
+    /// is returned and no fuel is consumed.
```

Comment:
Maybe saturating arithmetic is the right thing to do here instead of returning an error? Unsure.

I guess we could punt that saturate-or-not decision to the embedder since they can check `fuel_consumed` and compare it to the fuel added to clamp the amount of fuel to consume when calling this method. Might make sense to have a `fuel_remaining` method so they don't have to remember how much fuel was added and do that comparison themselves.

Reply:
I figure this is something we'll probably tweak over time yeah. The precise interfaces here aren't exhasutively considered and bike-shedded, rather they're mostly just situation-solving feature requests as we get them. For now I figured you could get `fuel_remaining` with `consume_fuel(0)`, but we can always add another path if necessary in the future too.